test/functional/unified-spec-runner/match.ts

```diff
@@ -254,48 +258,42 @@ export function expectErrorCheck(
   }
 
   if (expected.errorContains) {
-    if (error.message.includes(expected.errorContains)) {
-      throw new Error(
-        `Error message was supposed to contain '${expected.errorContains}' but had '${error.message}'`
-      );
-    }
+    expect(error.message).to.include(expected.errorContains);
   }
 
   if (!(error instanceof MongoError)) {
-    throw new Error(`Assertions need ${error} to be a MongoError`);
+    // if statement asserts type for TS, expect will always fail
+    expect(error).to.be.instanceOf(MongoError);
+    return;
   }
 
   if (expected.errorCode) {
```

Comment:
I think it'd be better to do `if (expected.errorCode != null)` here and in the other checks to allow for expectations with false-y values.